R/spliceHelperFunctions.R

```diff
@@ -480,27 +484,39 @@ dropGrangesListElementsByWidth <- function(grangesList, minWidth=5, cutStartEnd=
 #' @param stand strand
 #' @noRd
 selectStartExonFromRangesList <- function(range, strand){
-  posStrandRng <- range[which(strand == "+")]
-  negStrandRng <- range[which(strand == "-")]
-  negStrandRng <- negStrandRng[which(end(negStrandRng) == max(end(negStrandRng)))]
-  startRng <- c(posStrandRng[which(start(posStrandRng) == min(start(posStrandRng)))],
-                negStrandRng[which(end(negStrandRng) == max(end(negStrandRng)))])
-  startRng <- startRng[sort(order(startRng)[names(range)])]
-  return (startRng)
+  partitioning<-PartitioningByEnd(cumsum(pmin(elementNROWS(range),1)), names=NULL)
```

Comment:
would this here be the same `  partitioning<-PartitioningByEnd(1:length(range), names=NULL)` ?

Reply:
Yes, this is the same. I have changed it in the version with relist().